### PR TITLE
Add support for Localization for Wasm

### PR DESCRIPTION
**Pull request by jeromelaban**
## PR Type
What kind of change does this PR introduce?
- Feature

## What is the current behavior?
Resources identified using x:Uid are not loaded in Wasm.

## What is the new behavior?
Resources identified using x:Uid are not loaded in Wasm, using the resw (PRIResource) files as sources.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [x] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [x] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [x] Contains **NO** breaking changes

### CODE REVIEW

File: src/SourceGenerators/Uno.UI.Tasks/ResourcesGenerator/UnoPRIResourcesWriter.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+{
+	internal class UnoPRIResourcesWriter
+	{
+		internal static void Write(string language, Dictionary<string, string> resources, string actualTargetPath, string comment)
+		{
+			using (var file = File.OpenWrite(actualTargetPath))
+			{
+				using (var writer = new BinaryWriter(file, Encoding.UTF8))
+				{
+					writer.Write(new byte[] { 0x75, 0x6E, 0x6F }); // Magic
```
Comment (davidjohnoliver) [inline src/SourceGenerators/Uno.UI.Tasks/ResourcesGenerator/UnoPRIResourcesWriter.cs]: :D

File: src/Uno.UWP/ApplicationModel/Resources/ResourceLoader.cs [hunk 2]
```diff
@@ -38,5 +88,59 @@ public string GetString(string resource)
...
+			_resources.Clear();
+		}
+
+		internal static void ProcessResourceFile(string name, Stream input)
+		{
+			using (var reader = new BinaryReader(input))
+			{
+				if (!reader.ReadBytes(3).SequenceEqual(new byte[] { 0x75, 0x6E, 0x6F }))
+				{
+					throw new InvalidCastException($"The file {name} is not a resource file");
```
Comment (davidjohnoliver) [inline src/Uno.UWP/ApplicationModel/Resources/ResourceLoader.cs]: InvalidCastException?

File: src/SourceGenerators/Uno.UI.Tasks/ResourcesGenerator/UnoPRIResourcesWriter.cs (hunk 1 above)
Comment (MatFillion) [reply src/SourceGenerators/Uno.UI.Tasks/ResourcesGenerator/UnoPRIResourcesWriter.cs]: I don't understand :(

File: src/Uno.CrossTargetting.props [hunk 3]
```diff
@@ -106,7 +106,7 @@
 	</ItemGroup>
 	<MakeDir Directories="$(_TargetNugetFolder)" />
 
-	<Message Importance="high" Text="OVERRIDING NUGET PACKAGE CACHE: $(_TargetNugetFolder)" />
+	<Message Importance="high" Text="OVERRIDING NUGET PACKAGE CACHE: $(_TargetNugetFolder) and $(_TargetNugetRefFolder)" />
```
Comment (MatFillion) [inline src/Uno.CrossTargetting.props]: Is this fixing the problem that xaml files are not moved to the package cache also or it's unrelated?  (if it's related I'll apply that fix to all our repos)

File: src/Uno.UWP/ApplicationModel/Resources/ResourceLoader.cs [hunk 4]
```diff
@@ -38,5 +88,59 @@ public string GetString(string resource)
...
+		internal static void ClearResources()
+		{
+			_resources.Clear();
+		}
+
+		internal static void ProcessResourceFile(string name, Stream input)
+		{
+			using (var reader = new BinaryReader(input))
+			{
+				if (!reader.ReadBytes(3).SequenceEqual(new byte[] { 0x75, 0x6E, 0x6F }))
```
Comment (MatFillion) [inline src/Uno.UWP/ApplicationModel/Resources/ResourceLoader.cs]: the same magic?

File: src/SourceGenerators/Uno.UI.Tasks/ResourcesGenerator/UnoPRIResourcesWriter.cs (hunk 1 above)
Comment (agneszitte-nventive) [reply src/SourceGenerators/Uno.UI.Tasks/ResourcesGenerator/UnoPRIResourcesWriter.cs]: #jeromelaban's magic ⭐️

Comment (carldebilly) [reply src/SourceGenerators/Uno.UI.Tasks/ResourcesGenerator/UnoPRIResourcesWriter.cs]: Explanation: <http://www.asciitable.com/>
- `0x75`: `u`
- `0x6E`: `n`
- `0x6F`: `o`

That's just a header to confirm we're dealing with the right file type. Like `PK` (`0x4B50`) for a _zip_ file (PK for _Phil Katz_, founder of PKWare) or `MZ` (0x4D5A) for a MS-DOS executable file (MZ for _Mark Zbikowski_, a Microsoft achitect).  Following this logic, it should have been `0x4A4C` (for `JL`).

File: src/Uno.UWP/ApplicationModel/Resources/ResourceLoader.cs (hunk 4 above)
Comment (carldebilly) [reply src/Uno.UWP/ApplicationModel/Resources/ResourceLoader.cs]: That kind of header is often called a "magic header". So you're right.

File: src/Uno.CrossTargetting.props (hunk 3 above)
Comment (jeromelaban) [reply src/Uno.CrossTargetting.props]: It's unrelated, it's just about telling which reference folder is being updated.

**Pull request merged**